go-controller/pkg/ovn/pods_test.go

```diff
@@ -84,34 +85,34 @@ func (p pod) baseCmds(fexec *ovntest.FakeExec) {
 
 func (p pod) populateLogicalSwitchCache(fakeOvn *FakeOVN) {
 	Expect(p.nodeName).NotTo(Equal(""))
-	fakeOvn.controller.lsMutex.Lock()
-	defer fakeOvn.controller.lsMutex.Unlock()
-	fakeOvn.controller.logicalSwitchCache[p.nodeName] = []*net.IPNet{ovntest.MustParseIPNet(p.nodeSubnet)}
+	fakeOvn.controller.lsCache.Add(p.nodeName, []*net.IPNet{ovntest.MustParseIPNet(p.nodeSubnet)})
+	p.allocateMockIP(fakeOvn)
+}
+
+func (p pod) allocateMockIP(fakeOvn *FakeOVN) {
+	podIP := net.ParseIP(p.podIP)
+	fakeOvn.controller.lsCache.ReleaseIP(p.nodeName, &podIP)
```

Comment:
either this is supposed to be `AllocateIP` or else it really needs an explanatory comment

Reply:
Ok... so this is taking advantage of the fact that if you call `ReleaseIP` on the mock IPAM allocator, it will cause that IP to allocated next... This is super non-obvious. If you are taking advantage of behavior specific to the mock allocator, you should call some mock-allocator-specific function.

However, it also seems like you ought to be able to get rid of the mock allocator entirely, and just use the "real" allocator, and adjust the test cases to deal with the IPs that the real allocator would return instead.

Reply:
So the challenge with using a real allocator earlier was not having control over what IP gets allocated. With the contiguous allocator, I can revisit this knowing that we will likely have a .4 address assigned to the test pods. Makes sense.

Reply:
done. got rid of the mock allocator and tweaked the tests.